server/plugin/api.go

```diff
@@ -219,15 +219,17 @@ func (p *MatterpollPlugin) handleVote(vars map[string]string, request *model.Pos
 		return commandErrorGeneric, nil, errors.Wrap(err, "failed to save poll")
 	}
 
-	v, err := poll.GetVotedAnswer(userID)
+	hasAdminPermission, appErr := p.HasAdminPermission(poll, userID)
+	if appErr != nil {
+		p.API.LogWarn("Failed to check permission", "userID", userID, "error", appErr.Error())
+		hasAdminPermission = false
+	}
+	metadata, err := poll.GetMetadata(userID, hasAdminPermission)
 	if err != nil {
 		return commandErrorGeneric, nil, errors.Wrap(err, "failed to get voted answers")
 	}
-	p.API.PublishWebSocketEvent("has_voted", map[string]interface{}{
-		"user_id":       v.UserID,
-		"poll_id":       v.PollID,
-		"voted_answers": v.VotedAnswers,
-	}, &model.WebsocketBroadcast{UserId: userID})
+
+	p.API.PublishWebSocketEvent("has_voted", metadata.ToMap(), &model.WebsocketBroadcast{UserId: userID})
```

Comment:
0/5 Maybe make this a it's own method:
```golang
p.PublishPollMetadat(p *poll.Poll, userID string)
```
and move the code above also in this method

Reply:
Good point!